### PR TITLE
LP-1043: notification handling based on course start and end

### PR DESCRIPTION
**Pull request by asamolion**

### CODE REVIEW

File: edx_notifications/server/api/consumer.py [hunk 1]
```diff
@@ -183,9 +187,28 @@ def post(self, request):
             return JsonResponse({"message": "Invalid notification payload"}, status=status.HTTP_400_BAD_REQUEST)
 
         user_ids = User.objects.filter(username__in=usernames).values_list('id', flat=True)
+        community_url = notification_data['categoryData']['slug']
```
Comment (muhammadnadeem) [inline edx_notifications/server/api/consumer.py]: Can we write it as notification_data.get('categoryData', {}).get('slug', '') ?

File: edx_notifications/server/api/consumer.py [hunk 2]
```diff
@@ -183,9 +187,28 @@ def post(self, request):
...
+
+        try:
+            # notification should redirect to NodeBB if course hasn't started yet
+            course_start_date = DiscussionCommunity.objects.raw('SELECT co.id, co.start \
+                FROM course_overviews_courseoverview AS co INNER JOIN nodebb_discussioncommunity \
+                AS dc ON co.id=dc.course_id WHERE dc.community_url="{}" LIMIT 1'.format(community_url))[0].start
+            if course_start_date > datetime.now(utc):
+                notification_source = 'nodebb'
+        except IndexError:
+            # If community is purely NodeBB community then redirect to community
```
Comment (muhammadnadeem) [inline edx_notifications/server/api/consumer.py]: Please make these comments more meaningful.

**Review by muhammadnadeem — CHANGES_REQUESTED** [review]

File: edx_notifications/server/api/consumer.py (hunk 1 above)
Comment (asamolion) [reply edx_notifications/server/api/consumer.py]: Yes we can.

File: edx_notifications/server/api/consumer.py (hunk 2 above)
Comment (asamolion) [reply edx_notifications/server/api/consumer.py]: Done..

**Review by muhammadnadeem — APPROVED** [review]
Looks good to me

**Pull request merged**